File: c/1021.c
/*  
    @autor: Guiventura6;
    @data: 21/01/2022;
    @Problema: Banknotes and Coins;    
*/
#include <stdio.h>

int main(){
    
    int notes, coins, notesOf100, notesOf50, notesOf20, notesOf10, notesOf5, notesOf2; 
    int coin1, coin50, coin25, coin10, coin05, coin01, resto;    
    double amount;
    
    scanf("%lf", &amount);
    notes = amount;

    notesOf100 = notes / 100;
    resto = notes % 100;
    notesOf50 = resto / 50;
    resto = resto % 50;
    notesOf20 = resto / 20;
    resto = resto % 20;
    notesOf10 = resto / 10;
    resto = resto % 10;
    notesOf5 = resto / 5;
    resto = resto % 5;
    notesOf2 = resto / 2;
    //COINS
    coin1 = resto % 2;
    coins = (amount - notes) * 100; // Working with integers
    coin50 = coins / 50;
    resto = coins % 50;
    coin25 = resto / 25;
    resto = resto % 25;
    coin10 = resto / 10;
    resto = resto % 10;
    coin05 = resto / 5;
    coin01 = resto % 5;

    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n", notesOf100);
    printf("%d nota(s) de R$ 50.00\n", notesOf50);
    printf("%d nota(s) de R$ 20.00\n", notesOf20);
    printf("%d nota(s) de R$ 10.00\n", notesOf10);
    printf("%d nota(s) de R$ 5.00\n", notesOf5);
    printf("%d nota(s) de R$ 2.00\n", notesOf2);

    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n", coin1);
    printf("%d moeda(s) de R$ 0.50\n", coin50);
    printf("%d moeda(s) de R$ 0.25\n", coin25);
    printf("%d moeda(s) de R$ 0.10\n", coin10);
    printf("%d moeda(s) de R$ 0.05\n", coin05);
    printf("%d moeda(s) de R$ 0.01\n", coin01);
        
    return 0;
}

File: python/1021.py
''' 
    Autor: Guiventura6;
    Data: 06/03/2022;
    Problema: Banknotes and Coins;
    Resume:calculate the smallest possible number of notes and coins on which the value can be decomposed.
    obs: The considered notes are of 100, 50, 20, 10, 5, 2. The possible coins are of 
    1, 0.50, 0.25, 0.10, 0.05 and 0.01
''' 
if __name__ == '__main__':
    
    amount = float(input())
        
    notes_Of_100 = amount // 100
    rest = amount % 100
    notes_Of_50 = rest // 50
    rest = rest % 50
    notes_Of_20 = rest // 20
    rest = rest % 20
    notes_Of_10 = rest // 10
    rest = rest % 10
    notes_Of_5 = rest // 5
    rest = rest % 5
    notes_Of_2 = rest // 2
    rest =  rest % 2
    #Coins
    notes_Of_1 = rest // 1
    coins = (rest - notes_Of_1) * 100 # Working with integers
    coins_of_50 = coins // 50
    rest = coins % 50
    coins_of_25 = rest // 25
    rest = rest % 25
    coins_of_10 = rest // 10
    rest = rest % 10
    coins_of_05 = rest // 5
    coins_of_01 = rest % 5
    
    print('NOTAS:')
    print('{:.0f} nota(s) de R$ 100.00'.format(notes_Of_100))
    print('{:.0f} nota(s) de R$ 50.00'.format(notes_Of_50))
    print('{:.0f} nota(s) de R$ 20.00'.format(notes_Of_20))
    print('{:.0f} nota(s) de R$ 10.00'.format(notes_Of_10))
    print('{:.0f} nota(s) de R$ 5.00'.format(notes_Of_5))
    print('{:.0f} nota(s) de R$ 2.00'.format(notes_Of_2))
    print('MOEDAS:')
    print('{:.0f} moeda(s) de R$ 1.00'.format(notes_Of_1))
    print('{:.0f} moeda(s) de R$ 0.50'.format(coins_of_50))
    print('{:.0f} moeda(s) de R$ 0.25'.format(coins_of_25))
    print('{:.0f} moeda(s) de R$ 0.10'.format(coins_of_10))
    print('{:.0f} moeda(s) de R$ 0.05'.format(coins_of_05))
    print('{:.0f} moeda(s) de R$ 0.01'.format(coins_of_01))        
    